www/js/util.js

```diff
@@ -1642,7 +1682,53 @@ function trimChatBuffer() {
     return count;
 }
 
-function pingMessage(isHighlight) {
+function showMessage(message, nameMentioned) {
+    if (!FOCUSED) {
+        if(USEROPTS.desktop_notification != "never") {
+                        
+            //It is possible that the user has revoked the permission to show notifications
+            //since the last time they opened cytube.
+            if(Notification && Notification.permission !== "granted") {
+                //If they deny/ignore the permission, the USEROPTS is set back to never.
+                getNotificationPermission();
+            }
+
+            if(USEROPTS.desktop_notification == "always" || 
+               USEROPTS.desktop_notification == "onlyping" && nameMentioned) {
+                   
+               notificationTitle = "/r/" + CHANNEL.name;
+               iconLocation = "/favicon.ico";
+               
+               //There are two ways to send notifications,
+               //service worker only works over HTTPS, so let's try that first.
+               try {
+                    navigator.serviceWorker.register('/sw.js');
```

Comment:
Does this need to be done on every message or only once?  Also, consistency of single vs. double quotes.